app/models/protocol.rb

```diff
@@ -269,32 +276,20 @@ def validate_proxy_rights
     errors.add(:base, "All users must be assigned a proxy right") unless self.project_roles.map(&:project_rights).find_all(&:nil?).empty?
   end
 
-  def principal_investigators
-    project_roles.reject{|pr| !['pi', 'primary-pi'].include?(pr.role)}.map(&:identity)
-  end
-
   def primary_principal_investigator
     primary_pi_project_role.try(:identity)
   end
 
   def primary_pi_project_role
-    project_roles.detect { |pr| pr.role == 'primary-pi' }
-  end
-
-  def billing_managers
-    project_roles.reject{|pr| pr.role != 'business-grants-manager'}.map(&:identity)
+    project_roles.find_by(role: 'primary-pi')
   end
 
   def billing_business_manager_email
     billing_business_manager_static_email.blank? ?  billing_managers.map(&:email).try(:join, ', ') : billing_business_manager_static_email
   end
 
-  def coordinators
-    project_roles.select{|pr| pr.role == 'research-assistant-coordinator'}.map(&:identity)
-  end
-
   def coordinator_emails
-    coordinators.map(&:email).try(:join, ', ')
+    coordinators.pluck(:email).try(:join, ', ')
```

Comment:
Is the join needed here? An empty pluck should return [] which can be joined to ''